### DISCUSSION TITLE
Failure to Compile Due to Missing Header

### DISCUSSION THREAD

**Discussion by GomerOfDoom** [su2code/SU2#1529] [Q&A]
**Describe the bug**
Master branch of SU2 version 7.3.0 will not compile due to the following error:

`In file included from ../SU2_CFD/src/output/COutputFactory.cpp:30:`
`../SU2_CFD/src/output/../../include/output/COutput.hpp:50:10: fatal error: mel.hpp: No such file or directory`
`   50 | #include "mel.hpp"`
`        |          ^~~~~~~~~`
`compilation terminated.`

In commit: ab5ab6c79be812db9dad77ffb9a24b1d96e50733 by Pedro on December 30, 2021, the following code was added to the master branch:

`/*--- AD workaround for a cmath function not defined in CoDi. ---*/`
`namespace mel::internal {`
`inline su2double hypot(const su2double& a, const su2double& b) {`
`  return sqrt(a*a + b*b);`
`}`
`}`
`#include "mel.hpp"`

What is mel.hpp?

If this is a workaround for a deficiency in CoDiPack, should it have a pre-processor flag to disable it when CodiPack is not in use?

EDITED: To recreate this error, use OS's meson rather than meson.py.

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - C++ compiler and version: GCC 11.1.0
 - MPI implementation and version: OpenMPI 4.1.2
 - SU2 Version: 7.3.0

**Comment by GomerOfDoom** [comment]
After a lot of configuring and reconfiguring, it turns out the above issue occurs when the system's meson is used instead of the meson.py that ships with SU2. However, I'm not sure this is correct behavior. Shouldn't the choice of the meson executable script be irrelevant? In other words, shouldn't the configuration be dependent on meson.build and not on meson.py? Using system's installed meson should produce the same result as using the meson.py that ships with SU2.

**Comment by pcarruscag** [comment]
It's used to implement #1478, and it was merged into develop.
Are those mesons the same version? Because the meson from Ubuntu 18.04 can't even configure SU2, let alone build.
Our build instructions state that you should use the versions shipped with the code.

**Reply by TobiKattmann** [reply]
Just for completeness in this discussion: `mel` stands for Math Expression Library that parses a string with a mathematical expression and translates that to a real computation

**Comment by GomerOfDoom** [comment]
Hi Pedro, thanks for responding. I will admit that I wrote the initial bug report in a bit of a state of frustration! Sorry about that!

I can't remember the details, but at some point in the past, I had switched to my system's meson b/c the meson.py that shipped with SU2 wasn't properly picking up / configuring things on my machine (or maybe it was on a cluster?). I will go back to using the meson.py included with SU2. 

In the meantime, a quick question: why are the changes (extensions to download and install the externals) to SU2's meson.py instead of meson.build? Can't those same extensions be put into the meson.build file?

As for the merging, that was my mistake. I see now that the changes came into master only about 5 days ago! I think I misunderstood the labelling on github's history pages while browsing code.
